sopel/modules/spellcheck.py

```diff
@@ -1,55 +1,151 @@
 # coding=utf-8
+
 """
-spellcheck.py - Sopel spell check Module
-Copyright © 2012, Elad Alfassa, <elad@fedoraproject.org>
-Copyright © 2012, Lior Ramati
+spellcheck.py - Sopel spelling checker module
+Copyright © 2016, Alan Huang
+Copyright © 2019, dgw
 Licensed under the Eiffel Forum License 2.
-
 https://sopel.chat
-
-This module relies on pyenchant, on Fedora and Red Hat based system, it can be found in the package python-enchant
 """
 from __future__ import unicode_literals, absolute_import, print_function, division
+import aspell
+from sopel.module import commands, example, require_admin
+
+
+def setup(bot):
+    bot.memory['spellcheck_pending_adds'] = []
+
+
+def shutdown(bot):
+    try:
+        del bot.memory['spellcheck_pending_adds']
+    except KeyError:
+        pass
+
+
+@commands('scadd')
+@require_admin('I only trust admins to add words.')
+def add_command(bot, trigger):
+    """
+    Stage a word to be added to the bot's personal dictionary.
+    """
+    bot.memory['spellcheck_pending_adds'].append(trigger.group(2))
+    bot.say('Added "{0}". (To review pending words, use {1}scpending. '
+            'To commit changes, use {1}scsave.)'
+            .format(trigger.group(2), bot.config.core.help_prefix))
 
-try:
-    import enchant
-except ImportError:
-    enchant = None
-from sopel.module import commands, example
+
+@commands('scpending')
+def pending_command(bot, trigger):
+    """
+    List words that are waiting to be saved to the bot's personal dictionary.
+    """
+    bot.say('Ready to save: "{0}". (To remove a word before saving, use {1}scdel '
+            'word, or clear the list with {1}scclear.)'
```

Comment:
Can there be something here to indicate that `word` in `..., use .scdel word, or ...` is an argument, and not literally the word `word`?

```suggestion
            '<word>, or clear the list with {1}scclear.)'
```

Reply:
There can, certainly. I was going for consistency with the rest of the core modules, which generally don't delineate arguments in any way.

Reply:
Hmm, no problem them. I'd say consistency trumps, then.

Reply:
If it's worth fixing, let's fix them all in another PR. 😸